cdn/signedurls/signurl_test.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func Test_readKeyFile_wrongPath(t *testing.T) {
+	if _, err := readKeyFile("/non/existing/path"); err == nil {
+		t.Fatal("expected error")
+	}
+}
+
+func Test_readKeyFile_badKey(t *testing.T) {
+	f, err := ioutil.TempFile("", "cdnkey")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(f.Name())
+	f.Close()
+
+	badKey := `KuTcCjT2/4G2/jVQRFVZUA==` // contains '/'
+	if err := ioutil.WriteFile(f.Name(), []byte(badKey), 0600); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := readKeyFile(f.Name()); err == nil {
+		t.Fatal("expected error")
+	}
+}
+
+func Test_readKeyFile(t *testing.T) {
+	f, err := ioutil.TempFile("", "cdnkey")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(f.Name())
+	f.Close()
+
+	key := `nZtRohdNF9m3cKM24IcK4w==`
+	expected := []byte{0x9d, 0x9b, 0x51, 0xa2, 0x17, 0x4d, 0x17, 0xd9,
+		0xb7, 0x70, 0xa3, 0x36, 0xe0, 0x87, 0x0a, 0xe3}
+	if err := ioutil.WriteFile(f.Name(), []byte(key), 0600); err != nil {
+		t.Fatal(err)
+	}
+	b, err := readKeyFile(f.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !reflect.DeepEqual(b, expected) {
+		t.Fatalf("got=%v; expected=%v", b, expected)
+	}
+}
+
+func TestSignURL(t *testing.T) {
```

Comment:
This is great test. I'm not familiar with go test frameworks. Is it possible to pass the 3 test inputs so that when executed, it will show as if 3 separate tests were executed?

Reply:
This is a fairly common method (“table driven tests”) that just uses the golang vanilla testing package. The output of the error makes it easy to tell what was the case that’s executed.

Reply:
added TODO